hyperspy/_signals/signal1d.py

```diff
@@ -1207,33 +1207,38 @@ def hanning_taper(self, side='both', channels=None, offset=0):
                 nav_chunks = dc.chunks[:-1]
             zeros = da.zeros(nav_shape + (offset,),
                              chunks=nav_chunks + ((offset,),))
-        if side == 'left' or side == 'both':
-            if self._lazy:
-                tapered = dc[..., offset:channels + offset]
-                tapered *= np.hanning(2 * channels)[:channels]
-                therest = dc[..., channels + offset:]
-                thelist = [] if offset == 0 else [zeros]
-                thelist.extend([tapered, therest])
-                dc = da.concatenate(thelist, axis=-1)
-            else:
-                dc[..., offset:channels + offset] *= (
-                    np.hanning(2 * channels)[:channels])
-                dc[..., :offset] *= 0.
-        if side == 'right' or side == 'both':
-            rl = None if offset == 0 else -offset
-            if self._lazy:
-                therest = dc[..., :-channels - offset]
-                tapered = dc[..., -channels - offset:rl]
-                tapered *= np.hanning(2 * channels)[-channels:]
-                thelist = [therest, tapered]
-                if offset != 0:
-                    thelist.append(zeros)
-                dc = da.concatenate(thelist, axis=-1)
-            else:
-                dc[..., -channels - offset:rl] *= (
-                    np.hanning(2 * channels)[-channels:])
-                if offset != 0:
-                    dc[..., -offset:] *= 0.
+        try:
```

Comment:
What about checking if the dtype is float instead of using the ``try except`` approach?